front/app/utils/aggs/bucketKeyIsMissing.ts

```diff
@@ -4,13 +4,21 @@ import {
 } from 'utils/constants';
 import { AggBucket } from 'containers/SearchPage/Types';
 
-function bucketKeyIsMissing(bucket: AggBucket): boolean {
-  const { key, keyAsString } = bucket;
+export function bucketKeyStringIsMissing(key: string): boolean {
+  return key == STRING_MISSING_IDENTIFIER || key === DATE_MISSING_IDENTIFIER;
+}
+
+export function bucketKeyAsStringIsMissing(keyAsString: string): boolean {
   return (
-    key == STRING_MISSING_IDENTIFIER ||
     keyAsString === STRING_MISSING_IDENTIFIER ||
-    key === DATE_MISSING_IDENTIFIER ||
     keyAsString === DATE_MISSING_IDENTIFIER
   );
 }
+
+function bucketKeyIsMissing(bucket: AggBucket): boolean {
+  return (
+    bucketKeyStringIsMissing(bucket.key) ||
```

Comment:
i broke these out to make the comparison more useful in different parts for now, but we should probably be operating over these buckets in those places so we can better utilize the key as string functionality for rendering things like dates.